File: app/src/main/java/com/lifedawn/bestweather/room/dao/KmaAreaCodesDao.java
package com.lifedawn.bestweather.room.dao;

import androidx.room.Dao;
import androidx.room.Query;

import com.lifedawn.bestweather.room.dto.KmaAreaCodeDto;

import java.util.List;

@Dao
public interface KmaAreaCodesDao {
	@Query("SELECT * FROM weather_area_code_table WHERE latitude_seconds_divide_100 >= :latitude-0.18 AND latitude_seconds_divide_100 <= "
			+ ":latitude+0.18 AND longitude_seconds_divide_100 >= :longitude-0.18 AND longitude_seconds_divide_100 <= :longitude+0.18")
	List<KmaAreaCodeDto> getAreaCodes(double latitude, double longitude);
}

File: app/src/main/java/com/lifedawn/bestweather/data/local/room/dao/KmaAreaCodesDao.kt
package com.lifedawn.bestweather.data.local.room.dao

import androidx.room.Dao
import androidx.room.Query
import com.lifedawn.bestweather.data.local.room.dto.KmaAreaCodeDto
import kotlinx.coroutines.flow.Flow

@Dao
interface KmaAreaCodesDao {
    @Query(
        "SELECT * FROM weather_area_code_table WHERE latitude_seconds_divide_100 >= :latitude-0.15 AND latitude_seconds_divide_100 <= "
                + ":latitude+0.15 AND longitude_seconds_divide_100 >= :longitude-0.15 AND longitude_seconds_divide_100 <= :longitude+0.15"
    )
    fun getAreaCodes(latitude: Double, longitude: Double): Flow<List<KmaAreaCodeDto>>
}